crates/tactical-server/src/main.rs

```diff
@@ -0,0 +1,217 @@
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    if let Err(e) = rt.block_on(notify_server_ready(&args)) {
+        error!("Failed to notify SpacetimeDB: {}", e);
+        std::process::exit(1);
+    }
+
+    // Run Bevy app
+    App::new()
+        .add_plugins(MinimalPlugins)
+        .add_plugins(AdventureSimulatorNetPlugins)
+        .insert_resource(MissionConfig {
+            mission_id: args.mission_id.clone(),
+            scene_key: args.scene_key.clone(),
+            spacetimedb_url: args.spacetimedb_url.clone(),
+            spacetimedb_module: args.spacetimedb_module.clone(),
+            port: args.port,
+        })
+        .insert_resource(MissionState {
+            started_at: std::time::Instant::now(),
+            enemies_killed: 0,
+            committed: false,
+        })
+        .add_systems(Startup, setup_server)
+        .add_systems(Update, check_mission_timeout)
+        .run();
+}
+
+#[derive(Resource)]
+struct MissionConfig {
+    mission_id: String,
+    scene_key: String,
+    spacetimedb_url: String,
+    spacetimedb_module: String,
+    port: u16,
+}
+
+#[derive(Resource)]
+struct MissionState {
+    started_at: std::time::Instant,
+    enemies_killed: u32,
+    committed: bool,
+}
+
+fn setup_server(mut commands: Commands, config: Res<MissionConfig>) {
+    info!("=== Tactical Server Ready ===");
+    info!("Mission: {}", config.mission_id);
+    info!("Scene: {}", config.scene_key);
+
+    let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), config.port);
+
+    // Start Lightyear server with auto self-signed cert
+    commands.spawn(AdventureSimulatorServer {
+        addr,
+        protocol: ServerProtocol::WebTransport {
+            certificate: WebTransportCertificateSettings::AutoSelfSigned(Default::default()),
+        },
+        protocol_settings: ProtocolSettings::default(),
+    });
+
+    info!("Listening on port {} (WebTransport)", config.port);
+    info!("Will timeout in {} seconds", MISSION_TIMEOUT_SECS);
+}
+
+#[allow(deprecated)]
+fn check_mission_timeout(
+    config: Res<MissionConfig>,
+    mut state: ResMut<MissionState>,
+    mut exit: EventWriter<AppExit>,
+) {
+    let elapsed = state.started_at.elapsed();
+
+    if elapsed >= Duration::from_secs(MISSION_TIMEOUT_SECS) && !state.committed {
+        info!("Mission timeout, committing results...");
+        state.committed = true;
+
+        let success = state.enemies_killed > 0;
+        let xp_gained = (state.enemies_killed * 25) as i32;
+
+        let rt = tokio::runtime::Runtime::new().unwrap();
+        if let Err(e) = rt.block_on(commit_mission(&config, success, xp_gained)) {
+            error!("Failed to commit mission: {}", e);
+        }
+
+        info!("Shutting down");
+        exit.write(AppExit::Success);
+    }
+}
+
+/// Notify SpacetimeDB that we're ready
+async fn notify_server_ready(args: &Args) -> Result<(), String> {
+    let client = reqwest::Client::new();
```

Comment:
Oof, I though we would have a spacetime connection via a `WebSocket` (so we can also subscribe to changes later). I mean for the strategic layer I get the need for rest api, but why here ?